sales_team_invoiced_target_domain/tests/test_crm_team.py

```diff
@@ -0,0 +1,66 @@
+from odoo import Command, fields
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+class TestCrmTeamInvoiced(BaseCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.team = cls.env["crm.team"].create({"name": "Test Team"})
+        cls.team2 = cls.env["crm.team"].create({"name": "Test Team 2"})
+        cls.partner_id = cls.env["res.partner"].create({"name": "Test Partner"})
+
+    def create_account_move(self, team_id, payment_state_code, amount):
+        move = self.env["account.move"].create(
+            {
+                "move_type": "out_invoice",
+                "team_id": team_id.id,
+                "date": fields.Date.context_today(self.env.user),
+                "amount_untaxed_signed": amount,
+                "partner_id": self.partner_id.id,
+                "invoice_line_ids": [
+                    Command.create(
+                        {"name": "Test Line", "quantity": 1, "price_unit": amount}
+                    )
+                ],
+            }
+        )
+        move.action_post()
+        move.payment_state = payment_state_code
+        return move
+
+    def test_compute_invoiced_multiple_teams_with_all_payment_states(self):
+        payment_states = ["not_paid", "in_payment", "paid", "partial", "reversed"]
+        self.env.company.sales_team_invoiced_domain = str(
+            [("payment_state", "in", payment_states)]
+        )
+        for state in payment_states:
+            self.create_account_move(self.team, state, 100)
+            self.create_account_move(self.team2, state, 50)
+        (self.team + self.team2)._compute_invoiced()
+        self.assertEqual(self.team.invoiced, 500.0)
+        self.assertEqual(self.team2.invoiced, 250.0)
+
+    def test_compute_invoiced_empty_domain(self):
+        self.env.company.sales_team_invoiced_domain = "[]"
+        self.create_account_move(self.team, "paid", 100)
+        self.create_account_move(self.team2, "not_paid", 0)
+        (self.team + self.team2)._compute_invoiced()
+        expected_team_invoiced = sum(
+            self.env["account.move"]
+            .search(self.team._prepare_invoice_domain())
+            .mapped("amount_untaxed_signed")
+        )
+        expected_team2_invoiced = sum(
+            self.env["account.move"]
+            .search(self.team2._prepare_invoice_domain())
+            .mapped("amount_untaxed_signed")
+        )
+        self.assertEqual(self.team.invoiced, expected_team_invoiced)
+        self.assertEqual(self.team2.invoiced, expected_team2_invoiced)
```

Comment:
This test is not consistent. You’re not really testing anything, because you're just comparing the amount with the same data already present in the system.
It would be better to compare the team’s invoiced value with a specific expected value (not with the `sum` from `account.move`)

```suggestion
        self.assertEqual(self.team.invoiced, 100) # the specific expected value  
        self.assertEqual(self.team2.invoiced, 200) # the specific expected value  
```

Reply:
Yes, tests should be deterministic with fixed data, not to replicate the business code. For that, you have to create in setup a deterministic dataset.

Reply:
you're absolutely right, also, the move with not_paid status should result in 0 because it doesn't follow the core filter